classes/db/Db.php

```diff
@@ -567,10 +567,10 @@ public function executeS($sql, $array = true, $use_cache = true)
 		if ($use_cache && $this->is_cache_enabled && $array)
 		{
 			$this->last_query_hash = Tools::encryptIV($sql);
-			if (($result = Cache::getInstance()->get($this->last_query_hash)) !== false)
+			if (Cache::getInstance()->exists($this->last_query_hash) !== false)
```

Comment:
Hi,

Cache::getInstance()->get() also check if key exist, so please leave Cache::getInstance()->get() in order to avoid a second call to Cache,  but all others modifications are welcome.

Can you please modify your PR and rebase it ?
Thank you.

Best regards.